server/src/test/java/org/opensearch/ingest/AbstractBatchingProcessorTests.java

```diff
@@ -0,0 +1,135 @@
+            IngestDocumentPreparer.createIngestDocumentWrapper(3),
+            IngestDocumentPreparer.createIngestDocumentWrapper(4),
+            IngestDocumentPreparer.createIngestDocumentWrapper(5)
+        );
+        List<IngestDocumentWrapper> resultList = new ArrayList<>();
+        processor.batchExecute(wrapperList, resultList::addAll);
+        assertEquals(wrapperList, resultList);
+        ArgumentCaptor<List> argumentCaptor = ArgumentCaptor.forClass(List.class);
+        verify(processor, times(3)).internalBatchExecute(argumentCaptor.capture(), any());
+        assertEquals(wrapperList.subList(0, 2), argumentCaptor.getAllValues().get(0));
+        assertEquals(wrapperList.subList(2, 4), argumentCaptor.getAllValues().get(1));
+        assertEquals(wrapperList.subList(4, 5), argumentCaptor.getAllValues().get(2));
+    }
+
+    public void testBatchExecute_randomBatches() throws Exception {
+        int batchSize = randomIntBetween(2, 32);
+        int docCount = randomIntBetween(2, 32);
+        DummyProcessor processor = spy(new DummyProcessor(TAG, DESCRIPTION, batchSize));
+        List<IngestDocumentWrapper> wrapperList = new ArrayList<>();
+        for (int i = 0; i < docCount; ++i) {
+            wrapperList.add(IngestDocumentPreparer.createIngestDocumentWrapper(i));
+        }
+        List<IngestDocumentWrapper> resultList = new ArrayList<>();
+        processor.batchExecute(wrapperList, resultList::addAll);
+        assertEquals(wrapperList, resultList);
+        verify(processor, times(docCount / batchSize + (docCount % batchSize == 0 ? 0 : 1))).internalBatchExecute(
+            anyList(),
+            any()
+        );
+    }
+
+    public void testBatchExecute_defaultBatchSize() throws Exception {
+        DummyProcessor processor = spy(new DummyProcessor(TAG, DESCRIPTION, 1));
+        List<IngestDocumentWrapper> wrapperList = Arrays.asList(
+            IngestDocumentPreparer.createIngestDocumentWrapper(1),
+            IngestDocumentPreparer.createIngestDocumentWrapper(2),
+            IngestDocumentPreparer.createIngestDocumentWrapper(3)
+        );
+        List<IngestDocumentWrapper> resultList = new ArrayList<>();
+        processor.batchExecute(wrapperList, resultList::addAll);
+        for (int i = 0; i < wrapperList.size(); ++i) {
+            assertEquals(wrapperList.get(i).getSlot(), resultList.get(i).getSlot());
+            assertEquals(wrapperList.get(i).getIngestDocument(), resultList.get(i).getIngestDocument());
+            assertEquals(wrapperList.get(i).getException(), resultList.get(i).getException());
+        }
+        verify(processor, never()).internalBatchExecute(anyList(), any());
+        verify(processor, times(3)).execute(any(IngestDocument.class));
+    }
+
+    static class DummyProcessor extends AbstractBatchingProcessor {
+
+        protected DummyProcessor(String tag, String description, int batchSize) {
+            super(tag, description, batchSize);
+        }
```

Comment:
Minor simplification, and you can remove the constants at the top of the class:

```suggestion
        protected DummyProcessor(int batchSize) {
            super("tag", "description", batchSize);
        }
```